experimental/segmenter_lstm/src/structs.rs

```diff
@@ -0,0 +1,30 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+
+use ndarray::Array1;
+use ndarray::Array2;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// 'LstmData' is a struct that store a LSTM model. Its attributes are:
+/// `model`: name of the model
+/// `dic`: the grapheme cluster dictionary used to train the model
+/// `mat1`: the matrix associateed with embedding layer
+/// `mat2` - `mat4`: the matrices associated with forward LSTM layer (embedding to hunits, hunits to hunits, and bias respectively)
+/// `mat5` - `mat7`: the matrices associated with backward LSTM layer (embedding to hunits, hunits to hunits, and bias respectively)
+/// `mat8` - `mat9`: the matrices associated with output layer (weight and bias term respectiely)
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub struct LstmData {
+    pub model: String,
+    pub dic: HashMap<String, i16>,
```

Comment:
Sorry about the drive-by comment, but: Even the inline part of `String` is three words. That's 8 3-byte scalar values in UTF-8 or 12 such scalar values as UTF-16. What's the maximum length of the EGCs? It seems to me that storing the EGCs inline either as UTF-16 or UTF-8 in a fixed-length array with the length indicated by trailing zeros in the unused array slots would likely be more efficient both in terms of inline size and in terms of pointer chasing than `String`.

Reply:
This struct isn't intended to be the final / most efficient form of storage.  This is just to get data from Python into Rust.  I expect that when we hook this up to the proper DataProvider, we will spend more time making sure we have the most compact form of data.